enumerable.rb

```diff
@@ -0,0 +1,104 @@
+module Enumerable
+  def my_each
+    each do |i|
+      yield i
+    end
+  end
+
+  def my_each_with_index
+    count = 0
+    while count < length
+      yield(self[count], count)
+      count += 1
+    end
+  end
+
+  def my_select
+    filtered_array = []
+    each do |i|
+      filtered_array.push(i) if yield i
+    end
+    filtered_array
+  end
+
+  def my_all?
+    each do |i|
+      return puts false unless yield i
+    end
+    puts true
+  end
```

Comment:
```rb
2.6.5 :035 > true_block = proc { |num| num <= 8 }
 => #<Proc:0x00000000018dbc28@(irb):35> 
2.6.5 :036 > array.all?(&true_block)
 => true 
2.6.5 :037 > array.my_all?(&true_block)
true
 => nil 
2.6.5 :038 >
```

Reply:
```rb
2.6.5 :040 > true_array = [1, true, 'hi', []]
 => [1, true, "hi", []] 
2.6.5 :041 > true_array.all?
 => true 
2.6.5 :042 > true_array.my_all?
Traceback (most recent call last):
        7: from /usr/share/rvm/rubies/ruby-2.6.5/bin/irb:23:in `<main>'
        6: from /usr/share/rvm/rubies/ruby-2.6.5/bin/irb:23:in `load'
        5: from /usr/share/rvm/rubies/ruby-2.6.5/lib/ruby/gems/2.6.0/gems/irb-1.0.0/exe/irb:11:in `<top (required)>'
        4: from (irb):42
        3: from enumerable.rb:25:in `my_all?'
        2: from enumerable.rb:25:in `each'
        1: from enumerable.rb:26:in `block in my_all?'
LocalJumpError (no block given (yield))
2.6.5 :043 > 
```

Reply:
```rb
2.6.5 :045 > array
 => [5, 3, 1, 8, 5, 7, 6, 6, 0, 5, 7, 6, 8, 2, 4, 2, 6, 4, 4, 3, 0, 3, 2, 6, 8, 5, 0, 1, 5, 0, 6, 1, 6, 7, 4, 2, 4, 1, 4, 1, 6, 2, 8, 5, 2, 5, 7, 5, 7, 2, 8, 3, 2, 5, 4, 3, 8, 0, 8, 6, 3, 5, 6, 0, 3, 6, 7, 1, 4, 5, 6, 0, 7, 6, 3, 1, 3, 3, 6, 6, 3, 6, 6, 0, 6, 5, 8, 2, 8, 6, 0, 7, 4, 6, 1, 1, 2, 3, 4, 2] 
2.6.5 :046 > array.all?(Integer)
 => true 
2.6.5 :047 > array.my_all?(Integer)
Traceback (most recent call last):
        5: from /usr/share/rvm/rubies/ruby-2.6.5/bin/irb:23:in `<main>'
        4: from /usr/share/rvm/rubies/ruby-2.6.5/bin/irb:23:in `load'
        3: from /usr/share/rvm/rubies/ruby-2.6.5/lib/ruby/gems/2.6.0/gems/irb-1.0.0/exe/irb:11:in `<top (required)>'
        2: from (irb):47
        1: from enumerable.rb:24:in `my_all?'
ArgumentError (wrong number of arguments (given 1, expected 0))
2.6.5 :048 >
```

Reply:
```rb
2.6.5 :050 > words
 => ["dog", "door", "rod", "blade"] 
2.6.5 :051 > words.all?(/d/)
 => true 
2.6.5 :052 > words.my_all?(/d/)
Traceback (most recent call last):
        5: from /usr/share/rvm/rubies/ruby-2.6.5/bin/irb:23:in `<main>'
        4: from /usr/share/rvm/rubies/ruby-2.6.5/bin/irb:23:in `load'
        3: from /usr/share/rvm/rubies/ruby-2.6.5/lib/ruby/gems/2.6.0/gems/irb-1.0.0/exe/irb:11:in `<top (required)>'
        2: from (irb):52
        1: from enumerable.rb:24:in `my_all?'
ArgumentError (wrong number of arguments (given 1, expected 0))
2.6.5 :053 > 
```

Reply:
```rb
2.6.5 :055 > array
 => [5, 3, 1, 8, 5, 7, 6, 6, 0, 5, 7, 6, 8, 2, 4, 2, 6, 4, 4, 3, 0, 3, 2, 6, 8, 5, 0, 1, 5, 0, 6, 1, 6, 7, 4, 2, 4, 1, 4, 1, 6, 2, 8, 5, 2, 5, 7, 5, 7, 2, 8, 3, 2, 5, 4, 3, 8, 0, 8, 6, 3, 5, 6, 0, 3, 6, 7, 1, 4, 5, 6, 0, 7, 6, 3, 1, 3, 3, 6, 6, 3, 6, 6, 0, 6, 5, 8, 2, 8, 6, 0, 7, 4, 6, 1, 1, 2, 3, 4, 2] 
2.6.5 :056 > array.all?(3)
 => false 
2.6.5 :057 > array.my_all?(3)
Traceback (most recent call last):
        5: from /usr/share/rvm/rubies/ruby-2.6.5/bin/irb:23:in `<main>'
        4: from /usr/share/rvm/rubies/ruby-2.6.5/bin/irb:23:in `load'
        3: from /usr/share/rvm/rubies/ruby-2.6.5/lib/ruby/gems/2.6.0/gems/irb-1.0.0/exe/irb:11:in `<top (required)>'
        2: from (irb):57
        1: from enumerable.rb:24:in `my_all?'
ArgumentError (wrong number of arguments (given 1, expected 0))
2.6.5 :058 >
```

Reply:
> ```ruby
> 2.6.5 :035 > true_block = proc { |num| num <= 8 }
>  => #<Proc:0x00000000018dbc28@(irb):35> 
> 2.6.5 :036 > array.all?(&true_block)
>  => true 
> 2.6.5 :037 > array.my_all?(&true_block)
> true
>  => nil 
> 2.6.5 :038 >
> ```

This method works as expected
![tse_error](https://user-images.githubusercontent.com/46751363/87017751-472d6a00-c1c8-11ea-8db6-45db0ccda9ef.png)

Reply:
Kindly check the console result, I put in comment, please.
From the console, you can read that `true` is printed but nothing was returned.

**I compared what is returned by your code from what the original method returns**